### ISSUE TITLE
Integrating the new ethereumjs-block version

### ISSUE THREAD

**Issue by alcuadrado** [ethereumjs/ethereumjs-blockchain#123]
I'm opening this issue to plan how to integrate the new `ethereumjs-block` in here.

In particular, I'm interested to know if you consider that it should be done in a major or minor version. I suspect that it should be a major. @s1na @holgerd77

**Comment by alcuadrado** [comment]
I kept thinking about this, and maybe we should release major versions here and in `-block` simultaneously. 

If we are doing that, it may be good to finally promisify this library's API.

**Comment by holgerd77** [comment]
@alcuadrado @s1na Yes, I think this makes sense. We should generally give this a bit more time and just take this out of the initial v4 release. We have two libraries open on the switch anyhow (block and patricia tree) and we can then just take our time and add these with an upcoming `v4.1` release.

I will then now prepare a final v4 release on the VM - think it's really time that we get this out - see an upcoming associated PR with release notes for a review.
